File: app/src/app/core/models/principal.ts
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

export type Principal = {
  id: string;
  username: string;
  firstName: string;
  lastName: string;
  fullName: string;
  picture: string | null;
  language: string;
  timezone: string;
  isAdmin: boolean;
  siteUrl: string;
  siteName: string;
  siteVersion: string | null;
  siteRelease: string | null;
  siteTimezone: string;
};

File: api/src/main/kotlin/es/ubu/lsi/ubumonitorweb/domain/Principal.kt
/*
 * Este fichero forma parte de UBUMonitorWeb.
 *
 * @author Marcelo Verteramo Pérsico
 */

package es.ubu.lsi.ubumonitorweb.domain

import java.time.ZoneId

data class Principal(
  val id: Int,
  val username: String,
  val firstName: String,
  val lastName: String,
  val fullName: String,
  val picture: String?,
  val language: String,
  val timezone: ZoneId,
  val isAdmin: Boolean,
  val siteUrl: String,
  val siteName: String,
  val siteVersion: String?,
  val siteRelease: String?,
  val siteTimezone: ZoneId,
)
